### PR TITLE
A new easyblock for Scipion that handles version 2 and newer.

### PR DESCRIPTION
**Pull request by akesandgren**
No support for older versions.

### CODE REVIEW

File: easybuild/easyblocks/s/scipion.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
+##
+# Copyright 2018 Ghent University
```
Comment (boegel) [inline easybuild/easyblocks/s/scipion.py]: New, so make this `2019-2019`?

File: easybuild/easyblocks/s/scipion.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    def __init__(self, *args, **kwargs):
+        """Initialize Scipion-specific variables."""
+        super(EB_Scipion, self).__init__(*args, **kwargs)
+        # strip off 'scipion-*' part to avoid having everything in a subdirectory
+        self.unpack_options = '--strip-components=1'
+
+        # scipion.cfg file is in start_dir which isn't set until later
+        # so it gets initialized in configure_step instead.
+        self.cfgdir = ""
+        self.cfgfile = ""
```
Comment (boegel) [inline easybuild/easyblocks/s/scipion.py]: `self.cfgfile` is only used in `configure_step`, so no need for a class variable, just make it `cfgfile` in `configure_step`?

`self.cfgdir` is only used to define `self.cfgfile`, so no need for that at all, just use:

```python
cfgfile = os.path.join(self.builddir, 'config', 'scipion.conf')
```

in `configure_step`

File: easybuild/easyblocks/s/scipion.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        # scipion.conf needs to be configured with paths to all the
+        # dependencies, stupid but that's the way it works at the
+        # moment.
+
+        # Initialize config dir/file
+        self.cfgdir = os.path.join(self.builddir, 'config')
+        self.cfgfile = os.path.join(self.cfgdir, 'scipion.conf')
+
+        # Create the config files and then make the subtitutions
+        cmd = '%s ./scipion --config %s config' % (self.cfg['preconfigopts'], self.cfgfile)
```
Comment (boegel) [inline easybuild/easyblocks/s/scipion.py]: also pick up `configopts` here

File: easybuild/easyblocks/s/scipion.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+            'LINKERFORPROGRAMS': os.environ['CXX'],
+            'MPI_CC': os.environ.get('MPICC', 'UNKNOWN'),
+            'MPI_CXX': os.environ.get('MPICXX', 'UNKNOWN'),
+            'MPI_LINKERFORPROGRAMS': os.environ.get('MPICXX', 'UNKNOWN'),
+            # Set MPI_LIBDIR/INCLUDE/BINDIR to dummy values.
+            # Makes it easier to detect if they are still misused someplace.
+            # To be removed or changed to the real values later...
+            'MPI_LIBDIR': '/dummy',
+            'MPI_INCLUDE': '/dummy',
+            'MPI_BINDIR': '/dummy',
```
Comment (boegel) [inline easybuild/easyblocks/s/scipion.py]: @akesandgren These should be cleaned up?

**Comment by akesandgren** [comment]
@boegel Anything else I should fix in this PR?

**Comment by akesandgren** [comment]
Also, this superseedes #846

**Comment by akesandgren** [comment]
@boegel any more comments on this one??

File: easybuild/easyblocks/s/scipion.py [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+
+
+class EB_Scipion(ExtensionEasyBlock):
+    """Support for building/installing Scipion."""
+
+    def __init__(self, *args, **kwargs):
+        """Initialize Scipion-specific variables."""
+        super(EB_Scipion, self).__init__(*args, **kwargs)
+        # strip off 'scipion-*' part to avoid having everything in a subdirectory
+        self.unpack_options = '--strip-components=1'
```
[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyblocks/s/scipion.py:49]: @boegel I have not idea what Hound wants here...

[the remark this answers is not in the document]
Comment (ocaisa) [reply easybuild/easyblocks/s/scipion.py:49]: Synch with the develop branch, this is related to an outdated configuration of the hound

File: easybuild/easyblocks/s/scipion.py [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+        self.configure_step()
+        self.build_step()
+        self.install_step()
+
+    def sanity_check_step(self):
+        """Custom sanity check for Scipion."""
+
+        custom_paths = {
+            'files': [os.path.join('bin', 'scipion')],
+            'dirs': ['config', 'pyworkflow'],
```
[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyblocks/s/scipion.py]: Don't trick me like that.... it should be 'data' not 'scipion' which is an executable :-)

File: easybuild/easyblocks/s/scipion.py [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+        }
+        return super(EB_Scipion, self).sanity_check_step(custom_paths=custom_paths)
+
+    def make_module_req_guess(self):
+        """Custom guesses for environment variables (PATH, ...) for Scipion."""
+        guesses = super(EB_Scipion, self).make_module_req_guess()
+        guesses.update({'PATH': ['bin']})
```
[the remark this answers is not in the document]
Comment (akesandgren) [reply easybuild/easyblocks/s/scipion.py]: You're quite right. That was residue from the previous version of the easyblock.

**Comment by smoors** [comment]
closing-reopening

**Comment by smoors** [comment]
Going in, thanks @akesandgren!

**Pull request merged**